schema/schema_test.go

```diff
@@ -0,0 +1,175 @@
+package schema_test
+
+import (
+	"bufio"
+	"bytes"
+	"io/ioutil"
+	"testing"
+
+	"github.com/hpcloud/termui/schema"
+
+	term "github.com/cloudfoundry/cli/cf/terminal"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// TestSchema tests schema package
+func TestSchema(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "Schema")
+}
+
+var _ = Describe("Schema", func() {
+	Describe("Test if schema asks correctly for a string", func() {
+		Context("ask for string", func() {
+			It("should be Insert string value for /stringtest [required]", func() {
+
+				str := `{
+   "type":"object",
+   "properties":{
+      "stringtest":{
+         "type":"string"
+      }
+   },
+   "required":[
+      "stringtest"
+   ]
+}`
+
+				in := bytes.Buffer{}
+				out := bytes.Buffer{}
+				writer := bufio.NewWriter(&out)
```

Comment:
You don't actually need a `writer` here, since `bytes.Buffer` implements `io.Writer` (you can pass `&out` directly in all the places you currently pass `writer`).  Except the `writer.Flush()`, that just gets removed :)